components/usage/pkg/apiv1/usage.go

```diff
@@ -8,20 +8,53 @@ import (
 	context "context"
 
 	v1 "github.com/gitpod-io/gitpod/usage-api/v1"
+	"github.com/gitpod-io/gitpod/usage/pkg/db"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/timestamppb"
+	"gorm.io/gorm"
 )
 
 var _ v1.UsageServiceServer = (*UsageService)(nil)
 
 type UsageService struct {
+	conn *gorm.DB
 	v1.UnimplementedUsageServiceServer
 }
 
 func (us *UsageService) GetBilledUsage(ctx context.Context, in *v1.GetBilledUsageRequest) (*v1.GetBilledUsageResponse, error) {
-	// TODO(geropl) Dummy data for now
-	response := v1.GetBilledUsageResponse{}
-	return &response, nil
+	usageRecords, err := db.GetUsage(ctx, us.conn, db.AttributionID(in.GetAttributionId()))
+	if err != nil {
+		return nil, status.Error(codes.Internal, "unable to retrieve billed usage")
+	}
+
+	var billedSessions []*v1.BilledSession
+	for _, usageRecord := range usageRecords {
+		var endTime *timestamppb.Timestamp
+		if usageRecord.StoppedAt.Valid {
+			endTime = timestamppb.New(usageRecord.StoppedAt.Time)
+		}
+		billedSession := &v1.BilledSession{
+			AttributionId:  string(usageRecord.AttributionID),
+			UserId:         "",
+			TeamId:         "",
+			WorkspaceId:    "",
+			WorkspaceType:  "",
+			ProjectId:      "",
+			InstanceId:     usageRecord.InstanceID.String(),
+			WorkspaceClass: "",
+			StartTime:      timestamppb.New(usageRecord.StartedAt),
+			EndTime:        endTime,
+			Credits:        int64(usageRecord.CreditsUsed),
```

Comment:
Regarding the missing fields here, I'm not sure how many we want to store in the usage table ourselves, vs how much we want to obtain by joining with the existing workspace table.

The motivation for storing usage was so that we don't need to recompute it on every query to the usage API, but as long as we have the instanceIds, we should be able to recover the workspace details here with joins.

Reply:
In general I agree that this is the strategy we should follow. But WorkspaceClass and WorkspaceType should definitely in here, because they are relevant for the billing parts.
And regarding the hierachy-tuple TeamId-ProjectId-UserId-WorkspaceId: I think being able to directly correlate this entry (without the indirection over WorkspaceInstance) is valuable as well, and helps to "place this session on the map" if you will. I think we should have those, too.

Reply:
FYI We talked in sync, and agreed to wait for @easyCZ and his opinion. :+1:

Reply:
IMO, we should have first-class fields (these are mostly the ones we want to group-by in our views or are important enough to show in list views.

All the fields above meet this first-class definition so I would keem them as is. We should populate them directly from the usage report.

For any extra fields, we should call into WorkspaceInstance for details, or setup a service which pre-joins these for us.